### ISSUE TITLE
VA-date overflows on small screens

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/vets-design-system-documentation#1548]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

While using the `va-date` web component and testing it on a small screen (320 pixels wide), the width is not responsive. It will overflow.

<img width="557" alt="va-date web component rendered at 320 pixels wide showing the year input overflowing and going on screen" src="https://user-images.githubusercontent.com/136959/218179607-d0c9e407-7289-4361-b85a-ed82d091aa4c.png">

## What I expected to happen

The `va-date` component should be responsive. It could either shorten the month text to better fit on the screen, or move the year input onto a new line, etc.

## Reproducing
- Formation version: Current
- Device: (e.g. iPhone 8, Macbook) All
- Browser: (e.g. Firefox, IE 11) All

Steps to reproduce:

1. No need to log in
2. Go to [Form 526 start page](https://staging.va.gov/disability/file-disability-claim-form-21-526ez/start) (not the same as the screenshot)
3. Choose "Yes" as being still on active duty
4. A `va-date` will appear.
5. Open the dev tools, toggle the device toolbar and set the width to 320 px

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [x] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

### Details

- Discussed in [Slack](https://dsva.slack.com/archives/CBU0KDSB1/p1674589851200819)
- Work-around is to use `va-memorable-date` instead

**Comment by GnatalieH** [comment]
Just wanted to comment that the [USWDS Memorable date component](https://designsystem.digital.gov/components/memorable-date/) also is not responsive.

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/1548?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc

**Comment by humancompanion-usds** [comment]
To be clear, va-date-input is going to have to work below 320px. At some point it will likely need to stack the inputs.

@rsmithadhoc - Can you set the severity on this and work with Carol to get it into the current sprint if you feel this is an a11y-defect-0 or a11y-defect-1.
